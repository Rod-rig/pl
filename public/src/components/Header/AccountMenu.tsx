import {IconButton, Menu, MenuItem} from '@material-ui/core';
import AccountCircle from '@material-ui/icons/AccountCircle';
import * as React from 'react';
import {Link} from 'react-router-dom';
import {dict} from '../../dict';

interface IProps {
  userName: string;
  logout: () => void;
}

const AccountLink = (props: any) => <Link to='/account' {...props}/>;
const StatLink = (props: any) => <Link to='/stats' {...props}/>;

export class AccountMenu extends React.Component<IProps, {
  anchor: HTMLElement,
}> {
  constructor(props: IProps) {
    super(props);
    this.state = {
      anchor: null,
    };
    this.openMenu = this.openMenu.bind(this);
    this.closeMenu = this.closeMenu.bind(this);
  }

  public openMenu(event: React.MouseEvent<HTMLElement>) {
    this.setState({anchor: event.currentTarget});
  }

  public closeMenu() {
    this.setState({anchor: null});
  }

  public render() {
    const {anchor} = this.state;
    const open = Boolean(anchor);
    return (
      <React.Fragment>
        <IconButton
          onClick={this.openMenu}
          color="inherit"
        >
          <AccountCircle />
        </IconButton>
        <Menu
          anchorEl={anchor}
          anchorOrigin={{
            horizontal: 'right',
            vertical: 'top',
          }}
          transformOrigin={{
            horizontal: 'right',
            vertical: 'top',
          }}
          open={open}
          onClose={this.closeMenu}
        >
          <MenuItem disabled={true} divider={true}><em>{this.props.userName}</em></MenuItem>
          <MenuItem onClick={this.closeMenu} component={AccountLink}>{dict.header_menu_account_link}</MenuItem>
          <MenuItem onClick={this.closeMenu} component={StatLink}>{dict.header_menu_stat_link}</MenuItem>
          <MenuItem onClick={this.props.logout}>{dict.logout}</MenuItem>
        </Menu>
      </React.Fragment>
    );
  }
}
