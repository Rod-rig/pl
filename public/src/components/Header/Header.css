.header__toolbar {
    display: flex;
}

.header__link {
    text-decoration: none;
    margin-left: 1rem;
}
