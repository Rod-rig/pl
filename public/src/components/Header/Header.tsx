import MenuIcon from 'material-ui-icons/Menu';
import AppBar from 'material-ui/AppBar';
import IconButton from 'material-ui/IconButton';
import {withStyles, withTheme} from 'material-ui/styles';
import Toolbar from 'material-ui/Toolbar';
import Typography from 'material-ui/Typography';
import * as React from 'react';
import Logo from '../Logo/Logo';

const styles = (theme: any) => ({
  header: {
    marginBottom: theme.spacing.unit,
  },
  title: {
    marginLeft: theme.spacing.unit * 2,
  },
});

const Header = (props?: any) => {
  const {classes} = props;

  return (
    <AppBar position='static' className={classes.header}>
      <Toolbar className='header__toolbar'>
        <IconButton color='inherit' aria-label='Menu'>
          <MenuIcon/>
        </IconButton>
        <Logo/>
        <Typography className={classes.title} variant='title' color='inherit'>
          Title
        </Typography>
      </Toolbar>
    </AppBar>
  );
};

export default withStyles(styles)(withTheme()(Header));
