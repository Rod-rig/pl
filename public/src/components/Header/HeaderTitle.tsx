import {Typography} from '@material-ui/core';
import * as React from 'react';
import {HashRouter, Route, Switch} from 'react-router-dom';
import {routes} from '../../routeConfig';

const Title = (title: string) => {
  return (
    <Typography variant='h6' color='inherit'>
      {title}
    </Typography>
  );
};

export const HeaderTitle = () => {
  return (
    <HashRouter>
      <Switch>
        {
          routes.map((route, index: number) => (
            <Route
              key={index}
              exact={route.exact}
              component={Title.bind(this, route.title)}
              path={route.path}
            />
          ))
        }
      </Switch>
    </HashRouter>
  );
};
