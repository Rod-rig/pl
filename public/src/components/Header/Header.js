import React from 'react';
import Logo from '../Logo/Logo';

import './Header.css';

const Header = () => (
    <header className='header'>
        <Logo/>
        <div className='header__menu'>
            <a className='header__link' href=''>Predictions</a>
            <a className='header__link' href=''>Statistics</a>
        </div>
    </header>
);

export default Header;