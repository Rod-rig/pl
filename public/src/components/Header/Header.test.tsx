import {mount} from 'enzyme';
import * as React from 'react';
import {MemoryRouter} from 'react-router-dom';
import * as renderer from 'react-test-renderer';
import Header from './Header';

describe('Header', () => {
  const headerInRouter = <MemoryRouter><Header/></MemoryRouter>;
  const header = mount(headerInRouter);

  it('should have logo', () => {
    const logo = header.find('Logo');
    expect(logo).toHaveLength(1);
  });

  it('should render correctly', () => {
    const tree = renderer
      .create(headerInRouter)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
