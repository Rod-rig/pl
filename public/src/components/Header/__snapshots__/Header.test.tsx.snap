// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should render correctly 1`] = `
<header
  className="MuiPaper-root-9 MuiPaper-elevation4-15 MuiAppBar-root-1 MuiAppBar-positionStatic-5 MuiAppBar-colorPrimary-7 header"
>
  <div
    className="MuiToolbar-root-36 MuiToolbar-gutters-37 header__toolbar"
  >
    <button
      aria-label="Menu"
      className="MuiButtonBase-root-44 MuiIconButton-root-38 MuiIconButton-colorInherit-39"
      disabled={false}
      onBlur={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex="0"
      type="button"
    >
      <span
        className="MuiIconButton-label-43"
      >
        <svg
          aria-hidden="true"
          className="MuiSvgIcon-root-47"
          color={undefined}
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
          />
        </svg>
      </span>
      <span
        className="MuiTouchRipple-root-53"
      />
    </button>
    <div
      className="logo"
    >
      <a
        href="/"
        onClick={[Function]}
      >
        <img
          alt="logo"
          className="logo__img"
          src="https://avatars2.githubusercontent.com/u/11474330"
        />
      </a>
    </div>
    <h2
      className="MuiTypography-root-60 MuiTypography-title-66 MuiTypography-colorInherit-79"
      style={
        Object {
          "marginLeft": 16,
        }
      }
    >
      Title
    </h2>
  </div>
</header>
`;
